src/colp/components/agency_usetype_report.py

```diff
@@ -0,0 +1,83 @@
+import streamlit as st
+from src.constants import COLOR_SCHEME
+from abc import ABC
+import plotly.express as px
+
+
+class CountRecordsReport(ABC):
+    def __call__(self):
+        if self.data is None:
+            st.info(f"No Count of records by {self.y_axis_col}")
+            return
+        self.set_title()
+        self.plot()
+
+    def set_title(self):
+        by_agency = ""
+        middle_str = ""
+        by_usetype = ""
+        by_agency = "agency" if self.by_agency else ""
+        by_usetype = " use type" if self.by_usetype else ""
+        middle_str = " and" if self.by_agency and self.by_usetype else ""
+        st.subheader(
+            f"City owned and leased properties by {by_agency}{middle_str}{by_usetype}",
+            anchor="corrections-applied",
+        )
+
+    def plot(self):
+        slider_input = st.select_slider(
+            label=f"Most common {self.category_plural} to visualize",
+            options=range(1, self.data.shape[0] + 1),
+            value=(1, 12),
+        )
+        self.data.sort_values(by="count", ascending=False, inplace=True)
+        fig1 = px.bar(
+            self.data.iloc[
+                slider_input[0] : slider_input[1],
```

Comment:
should this be - 1 now since you're adding 1 up top?

Reply:
I thought that if the data had 100 records, we want the slider to let us choose 1 through 100, which is range (1, 101)

Reply:
Yes, but you are then slicing the dataset by an off-by-one, so say if you tell the range to start at 1, it will actually start at the 2nd record here, since the first one is 0. so i think you need
```
self.data.iloc[
slider_input[0] -1 : slider_input[1] - 1
```

Reply:
Ok right thanks I do need the minus one on the integer before the `:`. I don't think I want the minus after it though, if the user chooses the 100th out of 100 records we want `self.data.iloc[<start>:100,]` right?

Reply:
Nope - if they chose the last record (101 by your slider), it will actually cause an indexing out of bounds error because the largest index is 100.

Reply:
I think because `range` doesn't include the stop, `range(1, 101)` produces a the sequence 1,2,...,99,100

Reply:
oh interesting ok!